OpenEdXMobile/src/main/java/org/edx/mobile/view/AccountFragment.kt

```diff
@@ -39,51 +60,205 @@ class AccountFragment : BaseFragment() {
+    }
+
+    private fun initPersonalInfo() {
+        if (config?.isUserProfilesEnabled == false) {
+            binding.containerPersonalInfo.visibility = View.GONE
+            return
+        }
+        loginPrefs?.let { prefs ->
+            prefs.currentUserProfile?.let { profileModel ->
+                if (!profileModel.email.isNullOrEmpty()) {
+                    binding.tvEmail.text = ResourceUtil.getFormattedString(resources, R.string.profile_email_description, "email", profileModel.email)
+                } else {
+                    binding.tvEmail.visibility = View.GONE
+                }
 
-        environment?.analyticsRegistry?.trackScreenView(Analytics.Screens.ACCOUNT_SETTINGS)
+                if (!profileModel.username.isNullOrEmpty()) {
+                    binding.tvUsername.text = ResourceUtil.getFormattedString(resources, R.string.profile_username_description, "username", profileModel.username)
+                } else {
+                    binding.tvUsername.visibility = View.GONE
+                }
+
+                binding.tvLimitedProfile.visibility = if (profileModel.hasLimitedProfile) View.VISIBLE else View.GONE
+
+                prefs.profileImage?.let { imageUrl ->
+                    Glide.with(requireContext())
+                            .load(imageUrl.imageUrlMedium)
+                            .into(binding.profileImage)
+                }
+                        ?: run { binding.profileImage.setImageResource(R.drawable.profile_photo_placeholder) }
+            }
+            binding.containerPersonalInfo.visibility = View.VISIBLE
+            binding.containerPersonalInfo.setOnClickListener {
+                trackEvent(Analytics.Events.PERSONAL_INFORMATION_CLICKED, Analytics.Values.PERSONAL_INFORMATION_CLICKED)
+                environment?.router?.showUserProfile(requireActivity(), prefs.username ?: "")
+            }
+        } ?: run { binding.containerPersonalInfo.visibility = View.GONE }
     }
 
-    override fun onNewIntent(intent: Intent) {
-        super.onNewIntent(intent)
-        handleIntentBundle(intent.extras)
+    private fun updateWifiSwitch() {
+        val wifiPrefManager = PrefManager(requireContext(), PrefManager.Pref.WIFI)
+        binding.switchWifi.setOnCheckedChangeListener(null)
+        binding.switchWifi.isChecked = wifiPrefManager.getBoolean(PrefManager.Key.DOWNLOAD_ONLY_ON_WIFI, true)
+        binding.switchWifi.setOnCheckedChangeListener { _, isChecked ->
+            if (isChecked) {
+                wifiPrefManager.put(PrefManager.Key.DOWNLOAD_ONLY_ON_WIFI, true)
+                wifiPrefManager.put(PrefManager.Key.DOWNLOAD_OFF_WIFI_SHOW_DIALOG_FLAG, true)
+                trackEvent(Analytics.Events.WIFI_ON, Analytics.Values.WIFI_ON)
+            } else {
+                showWifiDialog()
+            }
+        }
+    }
+
+    private fun showWifiDialog() {
+        val newFragment = NetworkCheckDialogFragment.newInstance(getString(R.string.wifi_dialog_title_help),
```

Comment:
What if we change the variable name to `dialogFrgament`, what do you think?

Reply:
I meant to rename `newFragment` variable into `dialogFragment` or something more meaningful.

Reply:
updated